### PR TITLE
tests: kernel: Convert test case to use ztest

### PR DESCRIPTION
**Pull request by punitvara**
This patch reduces unnecessary output on console. Those things
are replaced by ztest framework APIs

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

File: tests/kernel/errno/src/0001-tests-kernel-Convert-test-case-to-use-ztest.patch [hunk 1]
```diff
@@ -0,0 +1,444 @@
+From c8556181b709816a7b47957772a9eaea65c938a3 Mon Sep 17 00:00:00 2001
```
Comment (nashif) [inline tests/kernel/errno/src/0001-tests-kernel-Convert-test-case-to-use-ztest.patch]: remove this file

**Review by nashif — CHANGES_REQUESTED** [review]
please split the commit into multiple commits, one for each test you are changing.

File: tests/kernel/errno/src/0001-tests-kernel-Convert-test-case-to-use-ztest.patch (hunk 1 above)
Comment (punitvara) [inline tests/kernel/errno/src/0001-tests-kernel-Convert-test-case-to-use-ztest.patch]: Done

**Comment by nashif** [comment]
Please add the test name into the title of the commit message and make them unique, i.e

tests: errno: ....
tests: critical: ...

File: tests/kernel/critical/src/Makefile [hunk 2]
```diff
@@ -1,4 +1,3 @@
 ccflags-y += -I${ZEPHYR_BASE}/tests/include
 include $(ZEPHYR_BASE)/tests/Makefile.test
-
-obj-y = critical.o
+obj-y = main.o test_critical_context.o
```
Comment (Vudentz) [inline tests/kernel/critical/src/Makefile]: I don't really get it why we need to split the test definition and tests into different files.

Comment (punitvara) [inline tests/kernel/critical/src/Makefile]: @Vudentz  Some test cases have their own name in src/ directory like critical test case has file name critical.c , mslab test case has file name slab.c. All these looks different and messy. If we can have one main file and other file (whatever name is) , all the test case would looks same and easy to understand structure for any one.

File: tests/kernel/errno/prj.conf [hunk 3]
```diff
@@ -1 +1,2 @@
 # nothing yet
```
Comment (nashif) [inline tests/kernel/errno/prj.conf]: remove that comment

File: tests/kernel/errno/src/test_errno_context.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2015 Wind River Systems, Inc.
```
Comment (nashif) [inline tests/kernel/errno/src/test_errno_context.c]: add new line for copyright with Intel.

File: tests/kernel/critical/src/Makefile (hunk 2 above)
Comment (nashif) [inline tests/kernel/critical/src/Makefile]: remove test_ prefix from files

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/errno/prj.conf (hunk 3 above)
Comment (punitvara) [inline tests/kernel/errno/prj.conf]: Done

File: tests/kernel/errno/src/test_errno_context.c (hunk 4 above)
Comment (punitvara) [inline tests/kernel/errno/src/test_errno_context.c]: Done

File: tests/kernel/critical/src/Makefile (hunk 2 above)
Comment (punitvara) [inline tests/kernel/critical/src/Makefile]: Done

Comment (nashif) [inline tests/kernel/critical/src/Makefile]: We do not need to split every testcase, splitting only has an advantage when you are testing different topics in one testcase and need the logical separation to make it easier to read and understand. Simple tests can be kept in one file.

**Review by nashif — APPROVED** [review]

**Pull request merged**